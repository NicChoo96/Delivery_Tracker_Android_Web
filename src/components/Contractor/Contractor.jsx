import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import './Contractor.scss';

/**
 * This component will render the contractor name, phone number and commpany name
 * @category UI Component
 * @subcategory General
 * @class Contractor
 * @param {string} contractorName contractor's name
 * @param {number} phoneNumber contractor's phone number
 * @param {string} companyName contractor's company name
 * @return {ReactComponent}
 */
const Contractor = ({ contractorName, phoneNumber, companyName }) => (
  <div className="contractor__card-width">
    <Card>
      <CardContent className="contractor__card-wrapper">
        <Typography color="textPrimary" component="h1">
          <span className="contractor_name">{contractorName}</span>
        </Typography>
        <Typography variant="caption" component="p">
          from <span className="contractor__company-name">{companyName}</span>
        </Typography>
        <Typography variant="body2" component="p">
          Mobile number: {phoneNumber}
        </Typography>
      </CardContent>
    </Card>
  </div>
);

export default Contractor;
