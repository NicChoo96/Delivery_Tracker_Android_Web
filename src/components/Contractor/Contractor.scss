.contractor__card-width {
  display: inline-block;
}

.contractor__card-wrapper {
  margin-right: 2em;
}

.contractor__name {
  font-weight: 700;
}

.contractor__company-name {
  font-style: italic;
}
