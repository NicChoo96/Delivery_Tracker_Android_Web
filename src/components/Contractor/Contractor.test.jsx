import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';

import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Contractor from './Contractor';

afterEach(cleanup);

describe('Contractor', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Contractor />, div);
  });

  test('test values of component props', () => {
    const testValues = {
      contractorName: 'test name',
      phoneNumber: 'test number',
      companyName: 'test company name',
    };
    const component = (
      <Contractor
        data-testid="contractor__testprops"
        contractorName={testValues.contractorName}
        phoneNumber={testValues.phoneNumber}
        companyName={testValues.companyName}
      />
    );
    Object.keys(component).map((key) => {
      return expect(component.props.key).toBe(testValues.key);
    });
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Contractor />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
