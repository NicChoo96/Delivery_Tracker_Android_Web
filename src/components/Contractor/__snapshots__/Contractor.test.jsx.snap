// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contractor matches snapshot 1`] = `
<div
  className="contractor__card-width"
>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root contractor__card-wrapper"
    >
      <h1
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
      >
        <span
          className="contractor_name"
        />
      </h1>
      <p
        className="MuiTypography-root MuiTypography-caption"
      >
        from 
        <span
          className="contractor__company-name"
        />
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        Mobile number: 
      </p>
    </div>
  </div>
</div>
`;
